File: Go/roman2int/13_test.go
package roman2int

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_romanToInt(t *testing.T) {
	tests := []struct {
		s    string
		want int
	}{
		{s: "III", want: 3},
		{s: "LVIII", want: 58},
		{s: "MCMXCIV", want: 1994},
	}
	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := romanToInt(tt.s)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/roman2int/13.test.ts
import romanToInt from './13';
import { describe, test, expect } from 'bun:test';

interface tt {
  s: string;
  want: number;
}

const tests: tt[] = [
  { s: 'III', want: 3 },
  { s: 'LVIII', want: 58 },
  { s: 'MCMXCIV', want: 1994 },
];

describe('roman to integer', () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = romanToInt(t.s);
      expect(result).toBe(t.want);
    });
  });
});
